R/utils.R

```diff
@@ -67,9 +67,20 @@ is_url <- function(x) {
 
 # Given the path to a file, return a file:// URL.
 file_url <- function(filename) {
-  if (is_windows()) {
-    paste0("file://", normalizePath(filename, mustWork = TRUE))
+  if (is_windows() && getRversion() < R_system_version("4.2.0")) {
+    enc2 <- identity
   } else {
-    enc2utf8(paste0("file:///", normalizePath(filename, winslash = "/", mustWork = TRUE)))
+    enc2 <- enc2utf8
   }
+  file <- sub("(.*)#[^#]*", "\\1", filename)
+  anchor_suffix <- sub(file, "", filename)
+  enc2(paste0(
+    "file:///",
+    normalizePath(
+      path = file,
+      winslash = if (is_windows()) "\\" else "/",
```

Comment:
I think this use of `winslash` can be removed, because it defaults to `"\\"` anyway, and it also is ignored on non-windows platforms.

 From the `normalizePath` help page:

> winslash: the separator to be used on Windows - ignored elsewhere. Must be one of `c("/", "\\")`.

(I can see that I originally added the conditional `"/"` code, but I have no recollection of why I put it there.)

Reply:
Indeed, that's why it's still there as I thought "there must be a reason" ^^

Reply:
Removed in 820814cde20f9f363d228c259ea44eb2d66e1f4b